app/src/main/java/at/aau/serg/activities/RegisterActivity.kt

```diff
@@ -32,45 +34,44 @@ class RegisterActivity : AppCompatActivity() {
         val username = findViewById<EditText>(R.id.editTextUsername).text
         val password = findViewById<EditText>(R.id.editTextPassword).text
 
-        val callback = object : Callback {
-            override fun onFailure(call: Call, e: IOException) {
-                runOnUiThread{
-                    Toast.makeText(this@RegisterActivity, e.message, Toast.LENGTH_SHORT).show()
-                }
-            }
+        val error = authentication.registerUser(username.toString(), password.toString(), CallbackCreator().createCallback(::onFailureRegister, ::onResponseRegister))
+        if(error != null){
+            Toast.makeText(this, error, Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun onFailureRegister(){
+        runOnUiThread{
+            Toast.makeText(this@RegisterActivity, R.string.registerFailed, Toast.LENGTH_SHORT).show()
+        }
+    }
 
-            override fun onResponse(call: Call, response: Response) {
-                if (!response.isSuccessful) {
-                    runOnUiThread{
-                        Toast.makeText(this@RegisterActivity, "Registration failed", Toast.LENGTH_SHORT).show()
-                    }
-                    return
-                }
-                val responseBody = response.body?.string()
-                try{
-                    val jsonObject = JSONObject(responseBody)
-                    val accessToken = jsonObject.getString("accessToken")
-                    val refreshToken = jsonObject.getString("refreshToken")
-                    StoreToken().storeTokens(accessToken, refreshToken, ContextWrapper(this@RegisterActivity), Secret())
-                }catch (e: JSONException) {
-                    e.printStackTrace()
-                    Toast.makeText(this@RegisterActivity, "Registration success", Toast.LENGTH_SHORT).show()
-                    this@RegisterActivity.startActivity(Intent(this@RegisterActivity, LoginActivity::class.java))
-                    return
-                }
-                this@RegisterActivity.startActivity(Intent(this@RegisterActivity, MainActivity::class.java))
+    private fun onResponseRegister( response: Response){
+        if (!response.isSuccessful) {
+            runOnUiThread{
+                Toast.makeText(this@RegisterActivity, R.string.registerFailed, Toast.LENGTH_SHORT).show()
             }
+            return
         }
-        val error = Authentication(HttpClient(getString(R.string.api_url))).registerUser(username.toString(), password.toString(), callback)
-        if(error != null){
-            Toast.makeText(this, error, Toast.LENGTH_SHORT).show()
+        val responseBody = response.body?.string()
+        try{
+            StoreToken(ContextWrapper(this@RegisterActivity), Secret()).storeTokenFromResponseBody(JSONObject(responseBody))
+        }catch (e: JSONException) {
+            e.printStackTrace()
+            Toast.makeText(this@RegisterActivity, "Registration success", Toast.LENGTH_SHORT).show()
```

Comment:
Should we also tell the user why he has been moved to the `LoginActivity`?

Reply:
In my opinion it is not neccessary that the user knows why there was a problem py parsing the response from the server

Reply:
Ok, then we leave it as is.